pkg/controller/oneagentapm/oneagentapm_controller.go

```diff
@@ -43,6 +43,7 @@ func Add(mgr manager.Manager) error {
 		dtcReconciler: &utils.DynatraceClientReconciler{
 			Client:          client,
 			UpdatePaaSToken: true,
+			UpdateAPIToken:  true,
```

Comment:
Technically this is true only when immutable images are used. For customers upgrading they would receive errors for the missing ApiToken even if it doesn't affect them.